src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+    #                and it is enabled if 'role_enabled' is true in the corresponding option set.
+    #     option_set and attribute: attribute in option set that must be substituted
+    #     line_fmt: the format to use when building the line
+    # An empty rule is valid and means that the keyword part must be
+    # written as is, using the line_fmt specified.
+
+    my $rule_id = 0;
+    foreach my $keyword (sort keys %$config_rules) {
+        my $rule = $config_rules->{$keyword};
+        $rule = '' unless defined($rule);
+        $rule_id++;
+
+        # Initialize parser_options for this rule according the default for this file
+        my $rule_parsing_options = {%{$parser_options}};
+
+        # Check if the keyword is prefixed by:
+        #     -  a '-': in this case the corresponding line must be unconditionally
+        #               commented out if it is present
+        #     -  a '?': in this case the corresponding line must be commented out if
+        #               it is present and the option is undefined
+        my $comment_line = 0;
+        if ($keyword =~ /^-/) {
+            $keyword =~ s/^-//;
+            $comment_line = 1;
+        } elsif ($keyword =~ /^\?/) {
+            $keyword =~ s/^\?//;
+            $rule_parsing_options->{remove_if_undef} = 1;
+            *$self->{LOG}->debug(2, "$function_name: 'remove_if_undef' option set for the current rule");
+        }
+
+        # Split different elements of the rule
+        ($rule, my $line_fmt, my $value_fmt) = split /;/, $rule;
+        unless ($line_fmt) {
+            $line_fmt = $LINE_FORMAT_DEFAULT;
+        }
+        my $value_opt;
+        if ($value_fmt) {
+            ($value_fmt, $value_opt) = split /:/, $value_fmt;
+        } else {
+            $value_fmt = LINE_VALUE_AS_IS;
+        }
+        unless (defined($value_opt)) {
+            # $value_opt is a bitmask. Set to 0 if not specified.
+            $value_opt = 0;
+        }
+
+
+        # If the keyword was "negated", remove (comment out) configuration line if present and enabled
+        if ($comment_line) {
+            *$self->{LOG}->debug(1, "$function_name: keyword '$keyword' negated, removing configuration line");
+            $self->_removeConfigLine($keyword, $line_fmt);
+            next;
+        }
+
+
+        # Parse rule if it is non empty
+        my $rule_info;
+        if ($rule ne '') {
+            *$self->{LOG}
+              ->debug(1, "$function_name: processing rule $rule_id (variable=>>>$keyword<<<, rule=>>>$rule<<<, fmt=$line_fmt)");
+            $rule_info = $self->_parse_rule($rule, $config_options, $rule_parsing_options);
+            next unless $rule_info;
+            *$self->{LOG}->debug(2, "$function_name: information returned by rule parser: " . join(" ", sort(keys(%$rule_info))));
+
+            if (exists($rule_info->{error_msg})) {
+                *$self->{LOG}->error("Error parsing rule >>>$rule<<<: " . $rule_info->{error_msg});
```

Comment:
No, this is intentional. I think there was a commend of @ned21 about this. As each rule is independent, if a rule causes an error, just process the next one. But log an error so that the component becomes a failed dependency. IMO, it would be dangerous to let other components think that the current one using CAF::RuleBasedEditor has succeeded.